src/v/cloud_storage/remote_segment.cc

```diff
@@ -25,7 +25,9 @@
 #include <seastar/core/loop.hh>
```

Comment:
It's mostly because of workflow. The changes are initially committed to `shadow-indexing-merged` and then extracted into this PR. It's easier to do on file level `git checkout shadow-indexing-merged -- src/v/cloud_storage/remote_segment.h` etc. Cherry-picking all that here would result in a very messy change set IMO.